src/ngResource/resource.js

```diff
@@ -714,11 +766,14 @@ angular.module('ngResource', ['ng']).
             var responseInterceptor = action.interceptor && action.interceptor.response ||
               defaultResponseInterceptor;
             var responseErrorInterceptor = action.interceptor && action.interceptor.responseError ||
-              undefined;
-            var hasError = !!error;
-            var hasResponseErrorInterceptor = !!responseErrorInterceptor;
+              $q.reject;
+            var successCallback = onSuccess ? function(val) {
+              onSuccess(val, response.headers, response.status, response.statusText);
+            } : undefined;
```

Comment:
Isn't `noop` better than `undefined` here? (and in line 773)

Reply:
It is better for these to be `undefined` since there is different behaviour in the call to `then(...)` function if `onError` is a defined function. But this only would make a difference if we returned the promise that is created at https://github.com/angular/angular.js/pull/16446/files#diff-67ae87630c0e91cba343677ed730daafR861.

As it stands `undefined` and `noop` have the same result below.